### ISSUE TITLE
Cross compiling

### ISSUE THREAD

**Issue by mtudan** [eclipse-cyclonedds/cyclonedds-cxx#163]
I tried to cross compile this repo, and similarly to the `cyclonedds` repo, it fails at the configuration stage with the following error:
```
| CMake Error at src/idlcxx/CMakeLists.txt:14 (add_library):
|   Target "idlcxx" links to target "CycloneDDS::idl" but the target was not
|   found.  Perhaps a find_package() call is missing for an IMPORTED target, or
|   an ALIAS target is missing?
```

The `cyclonedds` repo has the following pull request which fixes the issue:
https://github.com/eclipse-cyclonedds/cyclonedds/pull/948

@k0ekk0ek can we align to submit something similar in this repo as well?

**Comment by k0ekk0ek** [comment]
Of course. It's been a while since I cross compiled the C++ api (and that was without the current serializer), but I don't see a reason why it shouldn't work. I think the only thing we require for it to work with C++ is that we update `idlcxx_generate` with the same logic. I've provided some comments to @thijsmie's PR, but I also know there's some other things that require his attention. Best way forward would probably be to wait for the changes to land in cyclonedds and then apply the same ones in this repository. Feel free to give it a try yourself and open a PR, but let me know if you want me to help out. Would that work for you?

**Comment by mtudan** [comment]
@k0ekk0ek where can I contact you to discuss my open questions?

**Comment by k0ekk0ek** [comment]
Sorry for the late reply @mtudan. Well, if you want you can always drop me an email on the address in the Git logs, but maybe it's better if we simply discuss here(?) Anyway, @thijsmie 's changes have since been included and #183 kind-of describes the same issue we're discussing here. I'm hoping the `idlcxx_generate` changes will be picked up there. If not, I'll have a go at it. Shouldn't be too much work. Did you manage to make any progress since your last message? Let me know if I can help out.

**Comment by mtudan** [comment]
@k0ekk0ek I was using a forked version of this repo so forgot to open a PR. Anyway, here it is https://github.com/eclipse-cyclonedds/cyclonedds-cxx/pull/189, it also covers @ichernev work on https://github.com/eclipse-cyclonedds/cyclonedds-cxx/pull/183.
